internal/runner/operation_pipeline_step.go

```diff
@@ -29,67 +29,26 @@ func (r *Runner) executePipelineStepOp(
 	case *pb.Pipeline_Step_Exec_:
 		return r.executePipelineStepExec(ctx, log, job, project, kind.Exec)
 
+	case *pb.Pipeline_Step_Build_:
```

Comment:
Ah! I would actually never hit this. I would change `RunPipeline` on the server to transform this into a build job _there_ so we never have to spin up this superflous ODR/task.

Reply:
Ahhh ok, I'll look into making that work 👍🏻

Reply:
Got that working in [b4a036a](https://github.com/hashicorp/waypoint/pull/3459/commits/b4a036a2350837512f4d951f6ce907735f1ef319)